cray_product_catalog/catalog_update.py

```diff
@@ -283,7 +304,20 @@ def main():
     if VALIDATE_SCHEMA:
         validate_schema(data)
 
-    update_config_map(data, CONFIG_MAP, CONFIG_MAP_NAMESPACE)
+    PRODUCT_CONFIG_MAP = format_product_cm_name(CONFIG_MAP, PRODUCT)
+
+    LOGGER.debug("Splitting cray-product-catalog data")
+    main_cm_data, prod_cm_data = split_catalog_data(data)
+
+    if prod_cm_data and PRODUCT_CONFIG_MAP == '':
+        LOGGER.error("Not updating ConfigMaps because the provided product name is invalid: '%s'", PRODUCT)
+        raise SystemExit(1)
+
+    update_config_map(main_cm_data, CONFIG_MAP, CONFIG_MAP_NAMESPACE)
+
+    # If PRODUCT_CONFIG_MAP is not an empty string and prod_cm_data is not an empty dict
+    if prod_cm_data:
+        update_config_map(prod_cm_data, PRODUCT_CONFIG_MAP, CONFIG_MAP_NAMESPACE)
```

Comment:
It looks like this will end up inserting the data in this product-specific ConfigMap with the product name as a top-level key, which seems redundant. For example, for the COS product, this code will be called with `PRODUCT` set to `cos`. This will put some info like the following under `cray-product-catalog`:

```
csm:
  VERSION:
    images: []
    recipes: []
    configuration: {}
```

Then it will create a ConfigMap named `cray-product-catalog-cos`, and that will also contain `cos` as a top-level key:

```
cos:
  VERSION:
    component_versions: {}
```

Isn't this `cos` key at the top level redundant?

Reply:
To re-use the functionality without any change we have the product name as top-level key. If this is changed to remove the product name as top-level key in product ConfigMap then update, delete, and query logic need to change to incorporate this change. To keep the implementation simple we have kept this product name as top-level key.

Reply:
We discussed this further in our call today, and we decided it is okay to keep the product name as the top-level key for the following reasons:

- Simplifies the code
- Ensures the schema is more similar between the two config maps which might lead to easier migration.